File: 프로그래머스/unrated/172928. 공원 산책/공원 산책.py
def solution(park, routes):
    move = {'E': (0,1), 'W': (0,-1), 'S':(1,0), 'N':(-1,0)}
    h = len(park) -1
    w = len(park[0]) -1
    myPos = [0, 0]

    for i in range(len(park)):
        for j in range(len(park[0])):
            if park[i][j] == 'S':
                myPos = [i,j]
                break

    print(myPos)
    
    for i in routes:
        way, dis = i.split(' ')
        dis = int(dis)    
        saveX=myPos[1]
        saveY=myPos[0]
        
        for _ in range(dis):
            print(way, dis)
            myPos[0]+=move[way][0]
            myPos[1]+=move[way][1]
            print('move->', myPos)
            if myPos[1] > w or myPos[0] > h or myPos[0] < 0 or myPos[1] < 0:
                myPos = [saveY, saveX]
                break
            if park[myPos[0]][myPos[1]] == 'X':
                myPos = [saveY, saveX]
                break
        print('for End', myPos)
    return myPos

File: 프로그래머스/unrated/172928. 공원 산책/공원 산책.js
function solution(park, routes) {
    let x = 0, y = 0;
    
    for(let i=0; i < park.length; i++){
        for(let j=0; j < park[0].length; j++){
            if(park[i][j] === 'S'){
                x = j;
                y = i;
                break;
            }
        }
    }
    
    
    for(route of routes){
        const dir = route[0];
        const move = Number(route[2]);
        let flag = 0;
        
        if(dir === 'E'){
            for(let i=x+1; i <= x+move; i++){
                if(i === park[0].length){
                    flag = 1;
                    break;
                }
                if(park[y][i] === 'X') flag = 1;
            }
            if(flag === 0) x+=move;
        }else if(dir === 'W'){
            for(let i=x-1; i >= x-move; i--){
                if(i < 0){
                    flag = 1;
                    break;
                }
                if(park[y][i] === 'X') flag = 1;
            }
            if(flag === 0) x-=move;
        }else if(dir === 'S'){
            for(let i=y+1; i <= y+move; i++){
                if(i === park[0].length){
                    flag = 1;
                    break;
                }
                if(park[i][x] === 'X') flag = 1;
            }
            if(flag === 0) y+=move;
        }else if(dir === 'N'){
            for(let i=y-1; i >= y-move; i--){
                if(i < 0){
                    flag = 1;
                    break;
                }
                if(park[i][x] === 'X') flag = 1;
                
            }
            if(flag === 0) y-=move;
        }
    }
    
    return [y, x]
}